pkg/kubeadm/master/manifests.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubemaster
+
+import (
+	"bytes"
+	"encoding/json"
+	_ "fmt"
+	"os"
+	"path"
+
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	api "k8s.io/kubernetes/pkg/api/v1"
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+// Static pod definitions in golang form are included below so that `kubeadm
```

Comment:
> We should consider embedding yaml files in the binary and editing them on runtime with gotemplate.

I thought it was much better to generate them by using native data structures and serialisation, as that gives us less chances to mess things up, and eliminates any duplication. Why do you thing template could be better?

> Also as an extra feature, we could write them to a dir (`/etc/kubernetes/manifests`), and let the user edit them to their taste there.

Those are written to `/etc/kubernetes/manifests` on the host already, so they are accessible for the user to edit, but if kubelet will be running they don't have a chance to edit them really. I am not sure how much that's needed, we should just provide enough tunables, there aren't that many options and it's better to stick to best practice really. Also, as we are talking about full-blown self-hosting, this is one of the main things to go away, isn't it?